### ISSUE TITLE
"Game over" error aborting draft 

### ISSUE THREAD

**Issue by Modedrain** [dr4fters/dr4ft#833]
Firstly, thank you for an amazing website. 

Yesterday my friends and I were cube drafting. We were 4 people in total, drafting 5 packs, with the "leisurely" time setting. Somewhere in the middle of the fifth and last pack all of us were instantly booted from the draft with a red "game over" error text in red. I'll attach a screenshot. 

Obviously we have no idea what caused this, but speculate the draft might have some hidden time limit or similar? I'd say we were pretty much exactly an hour in when the error hit us. 

Would this be possible to fix? Is there any way we can avoid this in the future other than simply drafting more quickly? Thanks again! 

![IMG_20200118_121903](https://user-images.githubusercontent.com/59877168/72663023-28ef6a00-39ee-11ea-915c-19ac76cd748c.png)

**Comment by HerveH44** [comment]
Hi, 

Yes you're right about the time: we stop every game an hour after it started. Doing so we ensure that we don't have any dead game. 

We could increase the amount of time before the game is over. Maybe 2 hours could be fine?

**Comment by Modedrain** [comment]
Two hours would be great

**Comment by ZeldaZach** [comment]
I didn't know this was a thing, interesting. I'd bump it to 1.5hrs as a start.

**Comment by mbankim** [comment]
My friends and I experienced this issue at the very end of our draft despite the increased timeout (assuming that was fixed). I'm wondering why this timeout isn't based on player activity? We were literally last pack last pick and had no way to save our decks. Might want to reopen this issue since this could be a problem, especially with newer players.

**Comment by tooomm** [comment]
@mbankim how long was your session open?

**Comment by mixmix** [comment]
Yeah could be possible to time since last activity. Would be good to know how prevalent this big is before committing time and work

**Comment by mixmix** [comment]
Relevant files
https://github.com/dr4fters/dr4ft/search?q=Expires&type=

**Comment by mixmix** [comment]
Could perhaps change this logic

```
if (game.expires < now) { game.kill("game over"); }
```

So that it checks if `game.isActive` then run the `game.renew()` instead of `game.kill` to auto extend.

What we need to know is what counts as "active" for players. Probably we want to record when a game started, then auto extend until we reach some max (3 hours?) then kill the game

Ideally the ui would also tell people their game was going to expire when it was getting close

**Comment by mbankim** [comment]
@tooomm iirc it was about 90 minutes so it seems to be working as intended, though it's not a great user experience when the room closes

I think a warning in the UI is a good first step

**Comment by tooomm** [comment]
>it's not a great user experience when the room closes
>
>I think a warning in the UI is a good first step

Agreed!

The expanding time idea for active rooms could be an interesting solution.

**Comment by laxect** [comment]
Refresh the timer on activity is really a good solution. I and my friend just experienced this issue. So sad...
